chains/evm/calls/bridge.go

```diff
@@ -117,3 +174,67 @@ func PrepareSetDepositNonceInput(domainID uint8, depositNonce uint64) ([]byte, e
 	}
 	return input, nil
 }
+
+type ContractCallerClient interface {
+	CallContract(ctx context.Context, callArgs map[string]interface{}, blockNumber *big.Int) ([]byte, error)
+}
+
+func ProposalStatus(evmCaller ContractCallerClient, p *proposal.Proposal) (relayer.ProposalStatus, error) {
+	a, err := abi.JSON(strings.NewReader(consts.BridgeABI))
+	if err != nil {
+		return relayer.ProposalStatusActive, err
+	}
+	input, err := a.Pack("getProposal", p.Source, p.DepositNonce, p.Data)
+	if err != nil {
+		return relayer.ProposalStatusInactive, err
+	}
+
+	msg := ethereum.CallMsg{From: common.Address{}, To: &p.BridgeAddress, Data: input}
+	out, err := evmCaller.CallContract(context.TODO(), ToCallArg(msg), nil)
+	if err != nil {
+		return relayer.ProposalStatusInactive, err
+	}
+	type bridgeProposal struct {
+		ResourceID    [32]byte
+		DataHash      [32]byte
+		YesVotes      []common.Address
+		NoVotes       []common.Address
+		Status        uint8
+		ProposedBlock *big.Int
+	}
+	res, err := a.Unpack("getProposal", out)
+	if err != nil {
+		return relayer.ProposalStatusInactive, err
+	}
+	out0 := *abi.ConvertType(res[0], new(bridgeProposal)).(*bridgeProposal)
+	return relayer.ProposalStatus(out0.Status), nil
+}
+
+func idAndNonce(srcId uint8, nonce uint64) *big.Int {
+	var data []byte
+	data = append(data, big.NewInt(int64(nonce)).Bytes()...)
+	data = append(data, uint8(srcId))
+	return big.NewInt(0).SetBytes(data)
+}
+
+func ProposalVotedBy(evmCaller ContractCallerClient, by common.Address, p *proposal.Proposal) (bool, error) {
```

Comment:
It kinda seems to me the naming of this would indicate it would return the array of relayers which voted on the proposal (without looking at arguments).
Don't really have a better option (maybe `IsProposalVotedBy`), but it threw me a bit when I read the code.